File: fake-store-quarkus/src/main/java/org/arielzarate/interfaces/rest/dto/ProductResponse.java
package org.arielzarate.interfaces.rest.dto;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ProductResponse {
    private Integer id;
    private String title;
    private Double price;
    private String description;
    private String category;
    private String image;
    private RatingResponse rating;

    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class RatingResponse {
        private Double rate;
        private Integer count;
    }
}

File: api-fake-hexagonal/src/main/kotlin/com/arielZarate/hexagonal_example/interfaces/model/ProductResponse.kt
package com.arielZarate.hexagonal_example.interfaces.model

data class ProductResponse(
    val id: Int,
    val title: String,
    val price: Double,
    val description: String,
    val category: String,
    val image: String,
    val rating: RatingResponse
) {
    data class RatingResponse(val rate: Double, val count: Int)
}